src/transaction_builder.js

```diff
@@ -465,7 +466,7 @@ TransactionBuilder.prototype.sign = function (index, keyPair, redeemScript, hash
 
       // if we don't have a scriptType, assume pubKeyHash otherwise
       if (!input.scriptType) {
-        input.prevOutScript = bscript.pubKeyHashOutput(bcrypto.hash160(keyPair.getPublicKeyBuffer()))
+        input.prevOutScript = bscript.pubKeyHashOutput(bcrypto.hash160(kpPubKey))
         input.prevOutType = 'pubkeyhash'
```

Comment:
This will be a significant increase for `.sign` in the P2PKH case... @jprichardson lol